frontend/src/components/Worksheet/WorksheetStats.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import styled from 'styled-components';
+import chroma from 'chroma-js';
+import { useWorksheet } from '../../contexts/worksheetContext';
+import { ratingColormap } from '../../queries/Constants';
+import styles from './WorksheetStats.module.css';
+
+const StyledStatPill = styled.span<{ bgColor?: string }>`
+  background-color: ${({ theme, bgColor }) => bgColor ?? theme.surface[0]};
+`;
+
+const courseNumberColormap = chroma
+  .scale(['#63b37b', '#ffeb84', '#f8696b'])
+  .domain([4, 6]);
+const creditColormap = chroma
+  .scale(['#63b37b', '#ffeb84', '#f8696b'])
+  .domain([4, 5.5]);
+const workloadColormap = chroma
+  .scale(['#63b37b', '#ffeb84', '#f8696b'])
+  .domain([12, 24]);
+
+export default function WorksheetStats() {
+  const { courses, hidden_courses, cur_season } = useWorksheet();
+  const { courseCnt, credits, workload, rating } = courses.reduce(
+    (acc, c) =>
+      hidden_courses[cur_season]?.[c.crn]
+        ? acc
+        : {
+            courseCnt: acc.courseCnt + 1,
+            credits: acc.credits + (c.credits ?? 0),
+            workload: acc.workload + (c.average_workload ?? 0),
+            rating: acc.rating + (c.average_rating ?? 0),
+          },
+    { courseCnt: 0, credits: 0, workload: 0, rating: 0 },
+  );
+  const avgRating = courseCnt === 0 ? 0 : rating / courseCnt;
```

Comment:
i think we should make this the average of courses with actual ratings, so an average isn't lower if there is a new course. (like if you have 5 courses, 4 of them rated 1 2 3 4 and the 5th is a new course, your average should be 1+2+3+4/4 not 1+2+3+4+0/5) And same for average workload. Other than that (and the image im boutta comment) looks good :)